plugins/com.google.cloud.tools.eclipse.sdk.ui/src/com/google/cloud/tools/eclipse/sdk/ui/preferences/CloudSdkPreferenceArea.java

```diff
@@ -244,6 +280,11 @@ public void setFilterPath(File path) {
 
     @Override
     protected boolean doCheckState() {
+      if (useLocalSdk != null && !useLocalSdk.getSelection()) {
```

Comment:
ditto. We should directly check the `isManagedSdkFeatureEnabled()` flag.

Reply:
done